### DISCUSSION TITLE
Many Line Error When Installing Homebrew

### DISCUSSION THREAD

**Discussion by robfreed10** [Homebrew/discussions#295] [Everyday usage]
Hello all, 

When I go to install Homebrew, I get a massive error in my terminal referencing my /etc/sudoers file. It is weird because I am already the admin on my computer. I have also tried enabling root user and that did not work. If anyone knows the issue please help. I will share a screenshot of the error. Thanks in advance. 

Cheers, 

Robert Freed
![Screen Shot 2020-12-15 at 2 00 10 PM](https://user-images.githubusercontent.com/68922663/102260107-fa1f9380-3edd-11eb-9579-59b5602d9ad1.png)

**Comment by carlocab** [comment]
Looks like your `/etc/sudoers` file is borked. Not sure of the best way to fix it, but Google may help you.

**Reply by robfreed10** [reply]
I have done a ton of digging on google, but no matter what happens, that error constantly persists. Also, what do you mean by borked?

**Reply by carlocab** [reply]
Sorry. It means broken. What I think you want to do is to reset the `/etc/sudoers` file back into its default state. Unfortunately, you need `sudo` to do that, but `sudo` won't work with a broken `/etc/sudoers` file.

You're stuck in a catch-22, I think. The only way out might be to reformat your hard drive and reinstall your OS.

**Reply by carlocab** [reply]
Actually, it may not be that bad yet.

Can you do `sudo view /etc/sudoers`?

**Reply by robfreed10** [reply]
When I do that I get the same error as when I try to install Homebrew.

**Reply by carlocab** [reply]
Right. Ok. Catch-22 it is then. Sorry, I think you'll need to reformat your HD/SSD and reinstall macOS. But don't jump into that just yet; maybe someone else will have a better suggestion.

**Reply by robfreed10** [reply]
Appreciate your help, when you say that what do you mean by reinstall macOS. I am on Mojave 10.14, you believe I should upgrade to the most recent OS?

**Reply by SMillerDev** [reply]
Just updating won't reset the system configuration files. What might be possible is opening `/etc/sudoers` in textedit. If that's possible you could delete the offending lines.

**Reply by carlocab** [reply]
Not necessarily. You can, if you wish. Actually, maybe just reinstalling the OS without erasing everything may work too, in which case upgrading to any newer version of macOS (including Catalina) might fix your problem too.

**Reply by robfreed10** [reply]
@SMillerDev I have heard conflicting arguments for that. Some have said deleting those lines could be quite harmful to my computers memory. Also, when I do open my /etc/sudoers file, the lines where the errors occur are a very weird piece of text. But I am unsure if I should openly share my etc/sudoers file contents

**Reply by SMillerDev** [reply]
> I have heard conflicting arguments for that. Some have said deleting those lines could be quite harmful to my computers memory. 

Sudoers manages access rights so it has nothing to do with the computer memory.

**Reply by robfreed10** [reply]
So you believe just deleting that text will help? It is just a text file, so I guess it could.

**Comment by carlocab** [comment]
Here's what my `/etc/sudoers` looks like. Maybe it will help you fix your problem if you can, in fact, open it using TextEdit.
```
#
# Sample /etc/sudoers file.
#
# This file MUST be edited with the 'visudo' command as root.
#
# See the sudoers man page for the details on how to write a sudoers file.

##
# Override built-in defaults
##
Defaults        env_reset
Defaults        env_keep += "BLOCKSIZE"
Defaults        env_keep += "COLORFGBG COLORTERM"
Defaults        env_keep += "__CF_USER_TEXT_ENCODING"
Defaults        env_keep += "CHARSET LANG LANGUAGE LC_ALL LC_COLLATE LC_CTYPE"
Defaults        env_keep += "LC_MESSAGES LC_MONETARY LC_NUMERIC LC_TIME"
Defaults        env_keep += "LINES COLUMNS"
Defaults        env_keep += "LSCOLORS"
Defaults        env_keep += "SSH_AUTH_SOCK"
Defaults        env_keep += "TZ"
Defaults        env_keep += "DISPLAY XAUTHORIZATION XAUTHORITY"
Defaults        env_keep += "EDITOR VISUAL"
Defaults        env_keep += "HOME MAIL"

Defaults        lecture_file = "/etc/sudo_lecture"

##
# User alias specification
##
# User_Alias    FULLTIMERS = millert, mikef, dowdy

##
# Runas alias specification
##
# Runas_Alias   OP = root, operator

##
# Host alias specification
##
# Host_Alias    CUNETS = 128.138.0.0/255.255.0.0
# Host_Alias    CSNETS = 128.138.243.0, 128.138.204.0/24, 128.138.242.0
# Host_Alias    SERVERS = master, mail, www, ns
# Host_Alias    CDROM = orion, perseus, hercules

##
# Cmnd alias specification
##
# Cmnd_Alias    PAGERS = /usr/bin/more, /usr/bin/pg, /usr/bin/less

##
# User specification
##

# root and users in group wheel can run anything on any machine as any user
root            ALL = (ALL) ALL
%admin          ALL = (ALL) ALL

## Read drop-in files from /private/etc/sudoers.d
## (the '#' here does not indicate a comment)
#includedir /private/etc/sudoers.d
```

**Reply by robfreed10** [reply]
That is exactly what my file looks like. But after your last line, in my file that is where the error begins. However, I can not decipher what this means.

**Reply by robfreed10** [reply]
ALL=NOPASSWD:SETENV: /var/root/Applications/souter/souter

this is just one error.

**Reply by carlocab** [reply]
I would delete that. Or, at least, comment it out using `#`. Do you know how that got there?

You can probably also fix the error by checking `man sudoers` for the correct syntax.

**Reply by robfreed10** [reply]
No clue how it got there at all. I needed to download MongoDB for a class and that download required me to install Homebrew, which is how I got here.

**Reply by gromgit** [reply]
> /var/root/Applications/souter/souter

I suspect you've been infected by malware: https://labs.sentinelone.com/how-adload-macos-malware-continues-to-adapt-evade/

At this point, I'd:
* make sure I had a good backup
* reformat and reinstall macOS
* restore only the files I needed from the backup
